numpy/core/setup_common.py

```diff
@@ -104,6 +121,24 @@ def check_api_version(apiversion, codegen_dir):
         "strtoll", "strtoull", "cbrt", "strtold_l", "fallocate",
         "backtrace", "madvise"]
 
+OPTIONAL_STDFUNCS_SIGS = {}
+OPTIONAL_STDFUNCS_CALLS = {}
+
+for func in OPTIONAL_STDFUNCS:
+    set_sig(OPTIONAL_STDFUNCS_SIGS, OPTIONAL_STDFUNCS_CALLS, func, "double", ["double"])
+
+set_sig(OPTIONAL_STDFUNCS_SIGS, OPTIONAL_STDFUNCS_CALLS, "strtoull", "unsigned long long", ["const char*", "char**", "int"])
+set_sig(OPTIONAL_STDFUNCS_SIGS, OPTIONAL_STDFUNCS_CALLS, "strtoll", "long long", ["const char*", "char**", "int"])
+set_sig(OPTIONAL_STDFUNCS_SIGS, OPTIONAL_STDFUNCS_CALLS, "strtold_l", "long double", ["const char*", "char**", "int /* locale_t */"])
```

Comment:
`"int /* locale_t */"`
^^ It's a peculiarity of the way the wasm ABI handles structs that this works. If I'm allowed to `#include <xlocale.h>` then I would just use that.